pymc/tests/test_mixture.py

```diff
@@ -971,3 +972,249 @@ def test_broadcasting_in_shape(self):
+        assert_moment_is_expected(model, expected)
+
+    @pytest.mark.parametrize(
+        "weights, comp_dists, size, expected",
+        [
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (3,),
+                np.full(shape=(3,), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (11, 7),
+                np.full(shape=(11, 7), fill_value=2.8),
+            ),
+            (
+                # size = (11, 7) will be overwritten by (5, 3)
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5, size=(11, 7)), Normal.dist(6, 3, size=(11, 7))],
+                (5, 3),
+                np.full(
+                    shape=(5, 3),
+                    fill_value=2.8,
+                ),
+            ),
+        ],
+    )
+    def test_iterable_broadcasted_mixture_moments_with_size(
+        self, weights, comp_dists, size, expected
+    ):
+        with Model() as model:
+            Mixture("x", weights, comp_dists, size=size)
+        assert_moment_is_expected(model, expected)
+
+    @pytest.mark.parametrize(
+        "weights, comp_dists, size, expected",
+        [
+            (
+                np.array([0.4, 0.6]),
+                Normal.dist(mu=np.array([-2, 6]), sigma=np.array([5, 3])),
+                None,
+                2.8,
+            ),
+            (
+                np.tile(1 / 13, 13),
+                Normal.dist(-2, 1, size=(13,)),
+                None,
+                -2,
+            ),
+            (
+                np.array([0.4, 0.6]),
+                Normal.dist([-2, 6], 3),
+                None,
+                2.8,
+            ),
+        ],
+    )
+    def test_single_component_mixture_moments(self, weights, comp_dists, size, expected):
+        with Model() as model:
+            Mixture("x", weights, comp_dists, size=size)
+        assert_moment_is_expected(model, expected)
+
+    @pytest.mark.parametrize(
+        "weights, comp_dists, size, expected",
+        [
+            (
+                np.array([0.4, 0.6]),
+                [
+                    MvNormal.dist(mu=np.array([-1, -2]), cov=np.eye(2) * 0.3),
+                    MvNormal.dist(mu=np.array([3, 5]), cov=np.eye(2) * 0.8),
+                ],
+                None,
+                np.array(
+                    [1.4, 2.2],
+                ),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [
+                    Dirichlet.dist(a=np.array([2, 3, 5])),
+                    MvNormal.dist(mu=np.array([-10, 0, 10]), cov=np.eye(3) * 3),
+                ],
+                None,
+                np.array(
+                    [-5.92, 0.12, 6.2],
+                ),
+            ),
+        ],
+    )
+    def test_iterable_multivariate_mixture_moments(self, weights, comp_dists, size, expected):
```

Comment:
Also useful to add some cases with size for single components like you did for iterables

Reply:
I added 4 tests under `test_broadcasted_single_component_mixture_moments`!